ghga_datasteward_kit/s3_upload/entrypoint.py

```diff
@@ -0,0 +1,230 @@
+#!/usr/bin/env python3
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Contains functionality to actually run the S3 upload.
+"""
+
+import asyncio
+import base64
+import logging
+from pathlib import Path
+
+import typer
+from ghga_service_commons.utils.crypt import encrypt
+
+from ghga_datasteward_kit import models
+from ghga_datasteward_kit.s3_upload.config import Config, LegacyConfig
+from ghga_datasteward_kit.s3_upload.downloader import ChunkedDownloader
+from ghga_datasteward_kit.s3_upload.uploader import ChunkedUploader
+from ghga_datasteward_kit.s3_upload.utils import (
+    LOGGER,
+    HttpxClientState,
+    check_adjust_part_size,
+    check_output_path,
+    handle_superficial_error,
+    httpx_client,
+    objectstorage,
+)
+from ghga_datasteward_kit.utils import load_config_yaml, read_token
+
+
+async def shared_path(input_path: Path, alias: str, config: LegacyConfig):
```

Comment:
Return type not specified. Maybe rename and make docstring more specific.

Reply:
What would you rename this to?

Reply:
Maybe 
```suggestion
async def check_and_upload_content(input_path: Path, alias: str, config: LegacyConfig):
    """
    Check and upload encrypted file content. This also includes a verification of the upload by
    download the content again and perform a checksum validation.
    
    Returns:
        A tuple of the used uploader instance and the file size
    """
```